src/Components/Components/src/Auth/AuthenticationStateProvider.cs

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Components
+{
+    /// <summary>
+    /// Provides information about the authentication state of the current user.
+    /// </summary>
+    public abstract class AuthenticationStateProvider
+    {
+        /// <summary>
+        /// Gets an <see cref="AuthenticationState"/> instance that describes
+        /// the current user.
+        /// </summary>
+        /// <param name="forceRefresh">If true, instructs the provider to re-determine the user's authentication state, which can be an expensive operation. If false, the provider may reuse data cached for the current user.</param>
+        /// <returns>An <see cref="AuthenticationState"/> instance that describes the current user.</returns>
+        public abstract Task<AuthenticationState> GetAuthenticationStateAsync(bool forceRefresh);
```

Comment:
We might now choose not to have the `forceRefresh` param on the base class method, but rather make this part of the semantics of specific subclasses.

For example, the client-side `ServerAuthenticationStateProvider`, which works by making an HTTP to the server, could have its own `RefreshState` method that re-issues the request to the server and triggers `AuthenticationStateChanged` using the result.

Reply:
Since I can't think of any reason why the *framework* needs to request a refresh in a polymorphic way, this no longer needs to be part of the base class. Developers can still trigger a refresh on their own subclasses, via their own APIs, if they want.

So this is now done.